File: src/main/java/net/shadowfacts/forgelin/preloader/ForgelinPlugin.java
package net.shadowfacts.forgelin.preloader;

import net.shadowfacts.forgelin.Tags;
import net.shadowfacts.forgelin.coroutines.MinecraftClientDispatcher;

import cpw.mods.fml.relauncher.FMLLaunchHandler;
import cpw.mods.fml.relauncher.IFMLLoadingPlugin;

import java.util.Map;

//Java class because Kotlin is unsafe to use with IFMLLoadingPlugin (fplib deploader)
public final class ForgelinPlugin implements IFMLLoadingPlugin {
    @Override
    public String[] getASMTransformerClass() {
        return new String[] {
                Tags.ROOT_PKG + ".internal.asm.ASMTransformer"
        };
    }

    @Override
    public String getModContainerClass() {
        return null;
    }

    @Override
    public String getSetupClass() {
        return "net.shadowfacts.forgelin.preloader.ForgelinSetup";
    }

    @Override
    public void injectData(Map<String, Object> data) {
        if (FMLLaunchHandler.side().isClient()) {
            //Kotlin call is safe here, deploader has run.
            MinecraftClientDispatcher.setup();
        }
    }

    @Override
    public String getAccessTransformerClass() {
        return null;
    }
}

File: src/main/kotlin/net/shadowfacts/forgelin/preloader/ForgelinPlugin.kt
package net.shadowfacts.forgelin.preloader

import cpw.mods.fml.relauncher.IFMLLoadingPlugin
import cpw.mods.fml.relauncher.IFMLLoadingPlugin.MCVersion
import cpw.mods.fml.relauncher.IFMLLoadingPlugin.Name
import net.shadowfacts.forgelin.Tags

/**
 * @author shadowfacts
 */
@MCVersion("1.7.10")
@Name("Forgelin Core")
class ForgelinPlugin : IFMLLoadingPlugin {
    override fun getASMTransformerClass(): Array<String?> = arrayOf(Tags.ROOT_PKG + ".internal.asm.ASMTransformer")

    override fun getModContainerClass() = null

    override fun getSetupClass() = "net.shadowfacts.forgelin.preloader.ForgelinSetup"

    override fun injectData(data: Map<String?, Any?>?) {}

    override fun getAccessTransformerClass() = null
}